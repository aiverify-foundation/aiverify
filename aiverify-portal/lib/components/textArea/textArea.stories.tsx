import type { Meta, StoryObj } from '@storybook/react';
import { TextArea } from '.';

const meta: Meta<typeof TextArea> = {
  title: 'Components/Text Area',
  component: TextArea,
  parameters: {
    controls: {
      sort: 'requiredFirst',
      expanded: false,
    },
  },
  args: {
    disabled: false,
    resizeEnabled: false,
  },
  argTypes: {
    id: {
      table: {
        disable: true,
      },
    },
    name: {
      table: {
        disable: true,
      },
    },
    label: {
      control: 'text',
      description: 'Label of the textarea',
    },
    placeholder: {
      control: 'text',
      description: 'Placeholder text displayed when the input is empty',
    },
    disabled: {
      control: 'boolean',
      description: 'Disabled state of the text input',
      table: {
        type: {
          summary: 'boolean',
        },
        defaultValue: {
          summary: 'false',
        },
      },
    },
    resizeEnabled: {
      control: 'boolean',
      description: 'Enable resize of the text area',
      table: {
        type: {
          summary: 'boolean',
        },
        defaultValue: {
          summary: 'false',
        },
      },
    },
    containerStyles: {
      table: {
        disable: true,
      },
    },
    maxLength: {
      table: {
        disable: true,
      },
    },
    shouldFocus: {
      table: {
        disable: true,
      },
    },
    labelSibling: {
      table: {
        disable: true,
      },
    },
    style: {
      table: {
        disable: true,
      },
    },
    inputStyles: {
      table: {
        disable: true,
      },
    },
    labelStyles: {
      table: {
        disable: true,
      },
    },
    onChange: {
      table: {
        disable: true,
      },
    },
    onBlur: {
      table: {
        disable: true,
      },
    },
    onKeyDown: {
      table: {
        disable: true,
      },
    },
    onFocus: {
      table: {
        disable: true,
      },
    },
    labelClassName: {
      table: {
        disable: true,
      },
    },
    inputClassName: {
      table: {
        disable: true,
      },
    },
    error: {
      table: {
        disable: true,
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof TextArea>;

export const Default: Story = {
  args: {
    label: 'Address',
    placeholder: 'Enter your address',
  },
  render: (args) => (
    <div style={{ width: 400 }}>
      <TextArea
        {...args}
        labelClassName="dark:text-white"
        inputClassName="dark:border-primary-400 dark:focus:border-white"
      />
    </div>
  ),
};

export const ErrorTextarea: Story = {
  args: {
    label: 'Address',
    placeholder: 'Enter your address',
    error: 'Address is required',
  },
  render: (args) => (
    <div style={{ width: 400 }}>
      <TextArea {...args} />
    </div>
  ),
};
