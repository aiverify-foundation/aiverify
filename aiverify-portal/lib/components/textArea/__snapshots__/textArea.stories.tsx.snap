// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Text Area Default smoke-test 1`] = `
<div style="position: relative; width: 100%; height: 300px; display: flex; align-items: center; justify-content: center;">
  <div style="width: 400px;">
    <div class="_textArea_1mqdn_1">
      <label>
        <div class="_label_1mqdn_93 dark:text-white">
          <div>
            Address
          </div>
        </div>
        <textarea placeholder="Enter your address"
                  autocomplete="off"
                  class="dark:border-primary-400 dark:focus:border-white"
        >
        </textarea>
      </label>
    </div>
  </div>
</div>
`;

exports[`Components/Text Area ErrorTextarea smoke-test 1`] = `
<div style="position: relative; width: 100%; height: 300px; display: flex; align-items: center; justify-content: center;">
  <div style="width: 400px;">
    <div class="_textArea_1mqdn_1 _inputError_1mqdn_75">
      <label>
        <div class="_label_1mqdn_93">
          <div>
            Address
          </div>
        </div>
        <textarea placeholder="Enter your address"
                  autocomplete="off"
                  class
        >
        </textarea>
        <div class="_errorText_1mqdn_125">
          Address is required
        </div>
      </label>
    </div>
  </div>
</div>
`;
